### ISSUE TITLE
Supplier Returns: unable to delete Quantity to Return if the Quantity Available for Return is a decimal number

### ISSUE THREAD

**Issue by marthakeezy** [msupply-foundation/open-msupply#5121]
## What went wrong? 😲

I am playing around with Return Items, and it seems you are not able to delete the `Quantity to Return` number entered if the `Quantity Available for Return` is a decimal number. 

For example, I have 20.64 units available for return. 

- If I enter 20 to return, it works fine ✅ 
- If I enter 21 to return, the cell defaults to 20.64 units and I am no longer able to edit it 🚫 
- If I enter any decimal number to return (for example, 15.55 units to return), the cell defaults to 20.64 (why not 16?) and I am no longer able to edit it 🚫 

On a related note, yes to issues #4214 and #4179 - we should be able to enter decimal numbers in Returns and other areas of omSupply :)

Video of the behaviour is uploaded [here](https://drive.google.com/drive/folders/1FQ8f468ROLthxEbISRhuDQZV766497Rv)

## Expected behaviour 🤔

If the user is not able to enter decimal values in those cells (for now), there should be a clear behaviour to what the system does and also they should be able to still edit the cell if needed.

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to Replenishment -> Supplier Returns
2. Create a Return -> make sure it's for an item that has decimal values in stock. It might be easier to select an item, set quantity available to something decimal and then intentionally choose it for the return you are testing.
3. Try entering the scenarios described above (e.g. same number, whole number, number higher, decimal number) - then try to edit the cell and remove the number

## Your environment 🌱

- Open mSupply Version: `V2.3.0-RC8`
- Legacy mSupply Central Server Version: `V7.17.10`

- Platform: Tablet (apk)

- Database type: `SQLite3`

**Comment by roxy-dao** [comment]
The functionality has not been implemented yet as you have noted so this is expected behaviour for now until decimal packs is implemented for returns

**Comment by lache-melvin** [comment]
I can replicate this for stock lines with 2dp or more.

As Roxy said, we haven't implemented support for decimal packs sizes in returns/most of the app yet, so you shouldn't be able to enter decimal values, except that when you enter a large number, it defaults to max quantity available.

I don't think decimal packs support is on the immediate horizon, so I think we should get in a quick fix to allow a user to clear the field after it defaults to the decimal max (particularly on android, where highlighting the whole cell is hard)

Either - if total is 20.64, only allow return of 20, or allow backspacing of decimals as an "incomplete" value?

**Comment by marthakeezy** [comment]
Yes, until the functionality for decimal numbers is implemented, my only concern is that users should be able to backspace the value entered 🙏🏼

**Comment by andreievg** [comment]
Short term solution, max value on those inputs to be floored max available, we have other issues to solve it properly

**Comment by marthakeezy** [comment]
# 🧪 Testing



`V2.6.0-build-test-2`

- [ ] Go to Replenishment -> Supplier Returns
- [ ] Create a Return -> make sure it's for an item that has decimal values in stock. It might be easier to select an item, set quantity available to something decimal and then intentionally choose it for the return you are testing.
- [ ] Try entering different values in the ```Quantity to return``` input field (e.g same number, whole number, number higher, decimal number) 
- [ ] You should not be allowed to enter decimal numbers and numbers higher than available quantity to should round down to nearest integer
